### ISSUE TITLE
Misleading error in check "unable to parse config JSON"

### ISSUE THREAD

**Issue by lkhomenk** [pgbackrest/pgbackrest#667]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version:
2.08
2. PostgreSQL version:
10.6
3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:

4. Did you install pgBackRest from source or from a package?
crunchy-containers - crunchy-postgres with 2.08 pgbackrest

7. Describe the issue:
It more like a Improvement request. when running `pgbackrest check` - output I got was
```pgbackrest check
INFO: check command begin 2.08: --archive-timeout=600 --log-level-console=debug --log-level-file=debug --log-path=/tmp --pg1-path=/pgdata/a12 --repo1-path=/crunchy/ --repo1-s3-bucket=lkhomenk-s3 --repo1-s3-ca-file=/etc/pki/tls/certs/ca-bundle.crt --repo1-s3-endpoint=s3.eu-central-1.amazonaws.com --repo1-s3-key=<redacted> --repo1-s3-key-secret=<redacted> --repo1-s3-region=eu-central-1 --repo1-type=s3 --stanza=db
2019-02-07 11:50:13.291 P00  DEBUG:     config/load::cfgLoad: => void
2019-02-07 11:50:13.291 P00  DEBUG:     perl/exec::perlExec: (void)
ASSERT [025]: : unable to parse config JSON
ASSERT [025]: : option 'archive-async' does not exist
pgBackRest::Common::Exception=HASH(0x27d2d80).
 at /loader/0x204e9a8/pgBackRest/Main.pm line 12.
        pgBackRest::Main::__ANON__('pgBackRest::Common::Exception=HASH(0x27d2d80).\x{a}') called at -e line 0
```

Misleading was *"option 'archive-async' does not exist"* and even more - postgres log tells that it is required for some reason
```
ASSERT [025]: : unable to parse config JSON
--
  | ASSERT [025]: : option archive-async is required
  | pgBackRest::Common::Exception=HASH(0x1f470d0).
  | at /loader/0x1f2a918/pgBackRest/Main.pm line 12.
  | pgBackRest::Main::__ANON__('pgBackRest::Common::Exception=HASH(0x1f470d0).\x{a}') called at -e line 0
```
Issues was with key and secret key encrypted including "\n" and passed to pgbackrest which seems to break JSON made by perl backend
Original steps to reproduce:
1. encrypt key and secret like `echo key|base64` - create secret on Openshift\K8s. 
2. create crunchy-postgres container with PGBACKREST=true
3. in container console `pgbackrest check` - get errors

short steps:
1. `export PGBACKREST_S3_KEY=$(cat multiline_secret_file)` etc - this way it will hang `pgbackrest check`
 
I know that it's more our gap, but we were misled with that "async required" for a while

**Comment by lkhomenk** [comment]
just an idea - it might be useful to output current-config in json perl sees it.

**Comment by sfrost** [comment]
Thanks for the report, this does seem like something we should try to improve and perhaps we should also be removing any trailing whitespace, so that things like the 'cat' you had in your example work.  We'll discuss and update this with what we decide to do.

**Comment by dwsteele** [comment]
Yeah, the json renderer we use to pass options from C to Perl is not totally compliant.  Now that we have a compliant json renderer we'll switch over to using that.

I'm not sure that stripping whitespace is a good idea.  That could also have unintended consequences.

**Comment by dwsteele** [comment]
This has been fixed at 1f66bda02ec.

We don't do anything trim env variables, though.  I'm not sure whether this is a good idea -- and it might break current installations.
